src/App.js

```diff
@@ -20,12 +21,22 @@ function App() {
     console.log('new input value', newVal);
   };
 
+  const [inputValue, setInputValue] = useState('');
```

Comment:
Good use of hooks here

Reply:
I actually tried writing the new component as a function with hooks, but I kept getting errors about invalid hook calls. I couldn't figure out why, so I wrote it as a class instead.

Reply:
You should have a look into my pull request, I converted most of the class components to function components.